### PR TITLE
Fix `package` access modifier in XCBuild support

### PR DESCRIPTION
**Pull request by MaxDesiatov**
`package` access modifier was previously not supported in `swift build --build-system xcode`. This causes build issues when attempting to produce Universal binaries for SwiftPM, for example in this job https://ci.swift.org/job/swift-PR-source-compat-suite-debug-macos/1297/consoleFull.

The reason was that `-package-name` option was not added to `OTHER_SWIFT_FLAGS` in `PIFBuilder.swift`. Additionally, in llbuild support code we were shelling out to Swift Driver for every target to check whether `-package-name` is supported. Now with `-package-name` options calculation generalized across build systems, Swift Driver checks are done once per `BuildParameters` initialization, which reduces excessive shelling for llbuild.

Resolves rdar://120925895.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by neonichu** [comment]
> Additionally, in llbuild support code we were shelling out to Swift Driver for every target to check whether -package-name is supported. 

I don't think that's true since #7057? There's a static cache of computed flags, isn't there?

File: Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift [hunk 1]
```diff
@@ -422,7 +422,7 @@ public final class SwiftTargetBuildDescription {
     private func packageNameArgumentIfSupported(with pkg: ResolvedPackage, packageAccess: Bool) -> [String] {
```
Comment (neonichu) [inline Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift:422]: Can this use the the same code as the extension in `ResolvedPackage`? It seems like we duplicated that now.

File: Tests/BuildTests/BuildPlanTests.swift [hunk 2]
```diff
@@ -645,6 +645,38 @@ final class BuildPlanTests: XCTestCase {
         }
     }
 
+    #if os(macOS)
+    func testPackageNameFlagXCBuild() throws {
+        try XCTSkipIfCI() // test is disabled because it isn't stable, see rdar://118239206
```
Comment (neonichu) [inline Tests/BuildTests/BuildPlanTests.swift:650]: We shouldn't copy this from another test.

File: Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift (hunk 1 above)
Comment (MaxDesiatov) [reply Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift]: Good catch, that's addressed now.

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift/pull/70996

@swift-ci test linux

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Review by neonichu — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci clean test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/sourcekit-lsp/pull/1061
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/sourcekit-lsp/pull/1061
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/sourcekit-lsp/pull/1061
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift/pull/71793
https://github.com/apple/sourcekit-lsp/pull/1061
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift/pull/71793
https://github.com/apple/sourcekit-lsp/pull/1061
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift/pull/71793
https://github.com/apple/sourcekit-lsp/pull/1061
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift/pull/71793
https://github.com/apple/sourcekit-lsp/pull/1061
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Pull request merged**